be2-scala/src/main/scala/ch/epfl/pop/pubsub/PublishSubscribe.scala

```diff
@@ -95,9 +96,12 @@ object PublishSubscribe {
           /* partitioner port numbers */
           val portPipelineError = 0
           val portParamsWithMessage = 1
-          val portParamsWithChannel = 2
-          val portParamsWithMap = 3
-          val totalPorts = 4
+          val portSubscribe = 2
```

Comment:
Suggestion: maybe you can use define a Map , e.g. `portValues`. It still remains clear when you will use it in the code.